src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -1677,16 +1642,31 @@ void MacroAssembler::reinit_heapbase() {
   }
 }
 
-void MacroAssembler::movptr(Register Rd, address addr, int32_t &offset) {
-  int64_t imm64 = (int64_t)addr;
+void MacroAssembler::movptr(Register Rd, address addr, Register temp) {
+  int offset = 0;
+  movptr(Rd, addr, offset, temp);
+  addi(Rd, Rd, offset);
+}
+
+void MacroAssembler::movptr(Register Rd, address addr, int32_t &offset, Register temp) {
+  uint64_t uimm64 = (uint64_t)addr;
 #ifndef PRODUCT
   {
     char buffer[64];
-    snprintf(buffer, sizeof(buffer), "0x%" PRIx64, imm64);
+    snprintf(buffer, sizeof(buffer), "0x%" PRIx64, uimm64);
     block_comment(buffer);
   }
 #endif
-  assert((uintptr_t)imm64 < (1ull << 48), "48-bit overflow in address constant");
+  assert(uimm64 < (1ull << 48), "48-bit overflow in address constant");
+
+  if (temp == noreg) {
+    movptr_1(Rd, uimm64, offset);
+  } else {
+    movptr_2(Rd, uimm64, offset, temp);
+  }
+}
+
+void MacroAssembler::movptr_1(Register Rd, uint64_t imm64, int32_t &offset) {
```

Comment:
`movptr1` instead, to make it easily searchable with `is_movptr1_at`

Reply:
Fixed